File: backend/Api/LeagueSquadApi/Dtos/SquadMemberDtos.cs
namespace LeagueSquadApi.Dtos
{
    public record SquadMemberResponse(long SquadId, string Puuid, string? Role, string? Alias, DateTimeOffset CreatedAt, string GameName, string TagLine, string? Region);
    public record SquadMemberRequest(string Puuid, string? Role, string? Alias);
}

File: frontend/league-squad-web/src/types/SquadMemberDtos.ts
export interface SquadMemberResponse {
    squadId: number,
    puuid: string,
    role?: string,
    alias?: string, 
    createdAt: string, 
    gameName: string, 
    tagLine: string, 
    region?: string, 
}

// Split to AddSquadMemberRequest and GetSquadMemberRequest
// export interface AddSquadMemberRequest {
//     puuid: string,
//     role?: string, 
//     alias?: string
// }

export type AddSquadMemberRequest = Pick<SquadMemberResponse, "puuid" | "role" | "alias">

export type GetSquadMemberRequest = Pick<SquadMemberResponse, "squadId" | "puuid">
